examples/direct_fidelity_estimation.ipynb

```diff
@@ -0,0 +1,228 @@
+    "\n",
+    "Then, let's actually look at the measurements, i.e. $\\sigma_i$:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "rho_i=1.000\tsigma_i=1.000\tPauli:I\n",
+      "rho_i=1.000\tsigma_i=0.367\tPauli:Z(0)*X(1)*X(2)\n",
+      "rho_i=1.000\tsigma_i=0.564\tPauli:Z(1)*Z(2)\n",
+      "rho_i=-1.000\tsigma_i=-0.276\tPauli:Z(0)*Y(1)*Y(2)\n",
+      "rho_i=1.000\tsigma_i=0.651\tPauli:Z(0)\n",
+      "rho_i=1.000\tsigma_i=0.564\tPauli:X(1)*X(2)\n",
+      "rho_i=1.000\tsigma_i=0.367\tPauli:Z(0)*Z(1)*Z(2)\n",
+      "rho_i=-1.000\tsigma_i=-0.424\tPauli:Y(1)*Y(2)\n"
+     ]
+    }
+   ],
+   "source": [
+    "for trial_result in intermediate_results.trial_results:\n",
+    "    print('rho_i=%.3f\\tsigma_i=%.3f\\tPauli:%s' % (trial_result.pauli_trace.rho_i, trial_result.sigma_i, trial_result.pauli_trace.P_i))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "How are these measurements chosen? Since we had set `n_measured_operators=None`, all the measurements are used. If we had set the parameter to an integer, we would only have a subset to start from. We would then, as per the algorithm, sample from this set with replacement according to the probability distribution of $Pr(i)$ (for Clifford circuits, the probabilities are all the same, but for non-Clifford circuits, it means we favor more probable Pauli strings)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "What about the parameter `samples_per_term`? Remember that the code can handle both a sampler or use a simulator. If we use a sampler, then we can repeat the measurements `samples_per_term` times. In our case, we use a dense matrix simulator and thus we keep that parameter set to `0`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# How do we bound the variance of the fidelity when the circuit is Clifford?\n",
+    "Recall that the formula for DFE is:\n",
+    "$$F= \\sum _i Pr(i) . \\frac{\\sigma _i}{\\rho_i}$$\n",
+    "\n",
+    "But for Clifford circuits, we have $Pr(i) = \\frac{1}{d}$ and $\\rho_i = 1$ and thus the formula becomes:\n",
+    "$$F= \\frac{1}{d} \\sum _i \\sigma _i$$\n",
+    "\n",
+    "If we estimate by randomly sampling $N$ values for the indicies $i$ for $\\sigma_i$ we get:\n",
+    "$$\\hat{F} = \\frac{1}{N} \\sum_{j=1}^N \\sigma _{i(j)}$$\n",
+    "\n",
+    "Using the Bhatia–Davis inequality ([A Better Bound on the Variance, Rajendra Bhatia and Chandler Davis](https://www.jstor.org/stable/2589180)) and the fact that $0 \\le \\sigma_i \\le 1$, we have the variance of :\n",
```

Comment:
extreme nit: extraneous space before the colon ` :`

Reply:
Done.